### PR TITLE
Add FLO Presence 

### PR DESCRIPTION
**Pull request by gunyu1019**
**FLO is a South Korean music streaming service operated by SK Telecom. I am leaving Pull Requests to add this Presence.**
https://www.music-flo.com/ 

image
image
image
image

### CODE REVIEW

File: websites/F/FLO/presence.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "flo-logo",
+      buttons: [
+        {
+          label: "FLO 웹사이트",
+          url: "https://www.music-flo.com/"
+        }
+      ]
```
Comment (ririxidev) [inline websites/F/FLO/presence.ts:41]: image

File: websites/F/FLO/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "audio"
+  ],
+  "altnames": [
+    "플로"
+  ],
+  "url": "www.music-flo.com",
+  "description": {
+    "ko": "한국 음원 스트리밍 서비스(FLO)",
+    "en": "Korean Music Streaming Service (FLO)",
+    "nl": "Koreaanse muziekstreamingservice (FLO)"
```
Comment (ririxidev) [inline websites/F/FLO/dist/metadata.json:24]: I don't think that is their official description.

File: websites/F/FLO/presence.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+      presenceData.smallImageKey = "search";
+      presenceData.smallImageText = "앨범";
+      presenceData.details = "앨범";
+      presenceData.state = `${title} - ${artist}`;
+    }else if (location.pathname === "/browse") {
+      const title = document.querySelector(".chart_content_head>h4").textContent;
+
+      presenceData.smallImageKey = "search";
+      presenceData.smallImageText = title;
```
Comment (ririxidev) [inline websites/F/FLO/presence.ts:125]:
```suggestion
```

File: websites/F/FLO/presence.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      presenceData.smallImageKey = "search";
+      presenceData.smallImageText = "테마리스트";
+      presenceData.details = "테마리스트";
+      presenceData.state = title;
+    }else if (location.pathname.indexOf("/detail/album") === 0) {
+      const title = document.querySelector("p.title").textContent,
+       artist = document.querySelector("p.artist").textContent;
+
+      presenceData.smallImageKey = "search";
+      presenceData.smallImageText = "앨범";
```
Comment (ririxidev) [inline websites/F/FLO/presence.ts:118]:
```suggestion
```

File: websites/F/FLO/presence.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        presenceData.state = "공지사항";
+      else if (location.pathname.indexOf("/help/faq") === 0)
+        presenceData.state = "자주 묻는 문의";
+      else if (location.pathname.indexOf("/help/qna") === 0)
+        presenceData.state = "1:1 문의";
+    }else if (location.pathname.indexOf("/detail/channel") === 0) {
+      const title = document.querySelector("p.title").textContent;
+
+      presenceData.smallImageKey = "search";
+      presenceData.smallImageText = "테마리스트";
```
Comment (ririxidev) [inline websites/F/FLO/presence.ts:110]:
```suggestion
```

File: websites/F/FLO/presence.ts [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+          presenceData.details += "(앨범)";
+        else if (location.pathname === "/search/artist")
+          presenceData.details += "(아티스트)";
+        else if (location.pathname === "/search/theme")
+          presenceData.details += "(테마리스트)";
+        else if (location.pathname === "/search/lyrics")
+          presenceData.details += "(가사)";
+    }else if (location.pathname.indexOf("/new") === 0) {
+      presenceData.smallImageKey = "search";
+      presenceData.smallImageText = "최근 발매 음악";
```
Comment (ririxidev) [inline websites/F/FLO/presence.ts:90]:
```suggestion
```

File: websites/F/FLO/presence.ts [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+    }
+  } else { // Works when the player is not in use.
+    const {location} = window;
+    if (location.pathname === "/") 
+      presenceData.details = "메인";
+    else if (location.pathname.indexOf("/search") === 0) {
+        const keyword = getQuery();
+
+        presenceData.smallImageKey = "search";
+        presenceData.smallImageText = "검색";
```
Comment (ririxidev) [inline websites/F/FLO/presence.ts:74]:
```suggestion
```

File: websites/F/FLO/presence.ts [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+      playButtonActionId: actionType = DataRake(playButton.dataset),
+      title = player.querySelector("p.title").textContent,
+      artist = player.querySelector("p.artist").textContent,
+      allTime = player.querySelector(".time_all").textContent,
+      allHiddenTime = player.querySelector(".time_all").querySelector("span.hidden").textContent,
+      currentTime = player.querySelector(".time_current").textContent,
+      currentHiddenTime = player.querySelector(".time_current").querySelector("span.hidden").textContent,
+      nowDate = Date.now(),
+      allDate = getTimestamp(allTime, allHiddenTime),
+      currentDate = getTimestamp(currentTime, currentHiddenTime);
```
Comment (ririxidev) [inline websites/F/FLO/presence.ts:55]: Isn't time in unix timestamp or hh:mm:ss format? If yes, use our functions (`presence.getTimestamps()`, `presence.timestampFromFormat()`)

**Review by rxri — CHANGES_REQUESTED** [review]
Please run prettier over it too and check SV errors.

File: websites/F/FLO/dist/metadata.json (hunk 2 above)
Comment (Bas950) [reply websites/F/FLO/dist/metadata.json:24]: I think it's "음악을 들을수록 나를 더 닮아가는 나만의 FLO. 지금, 당신의 음악"

File: websites/F/FLO/presence.ts [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+      title = player.querySelector("p.title").textContent,
+      artist = player.querySelector("p.artist").textContent,
+      allTime = player.querySelector(".time_all").textContent,
+      allHiddenTime = player.querySelector(".time_all").querySelector("span.hidden").textContent,
+      currentTime = player.querySelector(".time_current").textContent,
+      currentHiddenTime = player.querySelector(".time_current").querySelector("span.hidden").textContent,
+      allDate = presence.timestampFromFormat(allTime.replace(allHiddenTime, "")),
+      currentDate = presence.timestampFromFormat(currentTime.replace(currentHiddenTime, ""));
+
+    [ presenceData.startTimestamp, presenceData.endTimestamp ] = presence.getTimestamps(currentDate, allDate);
```
Comment (ririxidev) [inline websites/F/FLO/presence.ts:44]:
```suggestion
    [, presenceData.endTimestamp] = presence.getTimestamps(currentDate, allDate);
```

**Review by rxri — CHANGES_REQUESTED** [review]
and prettier 😄

**Review by rxri — CHANGES_REQUESTED** [review]
still waiting for your prettified code with [prettier](https://prettier.io)

**Review by Slowlife01 — APPROVED** [review]

**Review by rxri — APPROVED** [review]

**Pull request merged**